patches/server/0852-Add-missing-structure-set-seed-configs.patch

```diff
@@ -0,0 +1,120 @@
+index 45f11284bf65081b3b2e8da85114efbe5efd5b42..ff3c8fcdae65d37e6cd4e388ee6cc047c24fe118 100644
+--- a/src/main/java/net/minecraft/world/level/levelgen/feature/MineshaftFeature.java
++++ b/src/main/java/net/minecraft/world/level/levelgen/feature/MineshaftFeature.java
+@@ -27,7 +27,13 @@ public class MineshaftFeature extends StructureFeature<MineshaftConfiguration> {
+ 
+     private static boolean checkLocation(PieceGeneratorSupplier.Context<MineshaftConfiguration> context) {
+         WorldgenRandom worldgenRandom = new WorldgenRandom(new LegacyRandomSource(0L));
++        // Paper start - add missing structure set seeds
++        if (context.chunkGenerator().mineshaftSeed != null) {
++            worldgenRandom.setLargeFeatureWithSalt(context.seed(), context.chunkPos().x, context.chunkPos().z, context.chunkGenerator().mineshaftSeed);
++        } else {
++        // Paper end
+         worldgenRandom.setLargeFeatureSeed(context.seed(), context.chunkPos().x, context.chunkPos().z);
++        } // Paper
+         double d = (double)(context.config()).probability;
+         return worldgenRandom.nextDouble() >= d ? false : context.validBiome().test(context.chunkGenerator().getNoiseBiome(QuartPos.fromBlock(context.chunkPos().getMiddleBlockX()), QuartPos.fromBlock(50), QuartPos.fromBlock(context.chunkPos().getMiddleBlockZ())));
+     }
+diff --git a/src/main/java/org/spigotmc/SpigotWorldConfig.java b/src/main/java/org/spigotmc/SpigotWorldConfig.java
+index b498b027b127996976a394e9a86cfc90f8a8ed3b..a8ea824af3601a06b5d3783160cd1b2b22e4914c 100644
+--- a/src/main/java/org/spigotmc/SpigotWorldConfig.java
++++ b/src/main/java/org/spigotmc/SpigotWorldConfig.java
+@@ -364,6 +364,15 @@ public class SpigotWorldConfig
+     public int mansionSeed;
+     public int fossilSeed;
+     public int portalSeed;
++    // Paper start - add missing structure set configs
++    public Integer mineshaftSeed;
++    public int buriedTreasureSeed;
++    public Long strongholdSeed;
++    private <N extends Number> N getSeed(String path, java.util.function.Function<String, N> toNumberFunc) {
++        final String value = this.getString(path, "default");
++        return org.apache.commons.lang3.math.NumberUtils.isParsable(value) ? toNumberFunc.apply(value) : null;
++    }
++    // Paper end
+     private void initWorldGenSeeds()
+     {
+         this.villageSeed = this.getInt( "seed-village", 10387312 );
+@@ -381,6 +390,11 @@ public class SpigotWorldConfig
+         this.mansionSeed = this.getInt( "seed-mansion", 10387319 );
+         this.fossilSeed = this.getInt( "seed-fossil", 14357921 );
+         this.portalSeed = this.getInt( "seed-portal", 34222645 );
++        // Paper start - add missing feature seed configs
++        this.mineshaftSeed = this.getSeed("seed-mineshaft", Integer::parseInt);
++        this.buriedTreasureSeed = this.getInt("seed-buried-treasure", 10387320); // def taken from BuriedTreasureFeature#RANDOM_SALT field
```

Comment:
Maybe add a diff on update to that field? Seems to already be gone in the latest snapshot though, so I'm not sure how long that'd last.

Reply:
You know, its not super clear that this can be changed in the latest snapshot at all. The reason you can't define a seed for buried treasure the same way you can for other structure sets, is that the default `spacing` and `separation` values are `1` and `0` which means the salt used has 0 effect because the random will *always* return 0. (the exclusive upper bound is `spacing - separation`. So the salt has no effect. there doesn't seem to be a replacement for it in the snapshots, no second random check.